lib/concurrent/synchronization/immutable_struct.rb

```diff
@@ -0,0 +1,67 @@
+module Concurrent
+  module Synchronization
+    # Similar to Struct but the fields are immutable and always visible (like Java final fields)
+    # @example
+    #   Person = ImmutableStruct.with_fields :name, :age
+    #   Person.new 'John Doe', 15
+    #   Person['John Doe', 15]
+    #   Person['John Doe', 15].members # => [:name, :age]
+    #   Person['John Doe', 15].values  # => ['John Doe', 15]
+    class ImmutableStruct < Synchronization::Object
```

Comment:
Are we using `ImmutableStruct` anywhere? I'm a big fan of the concept (and I've built similar constructs in other projects) but I can't seem to find a place where we use it.